packages/styles/scss/components/data-table/_data-table.scss

```diff
@@ -205,7 +205,7 @@
 
   .#{$prefix}--data-table td,
   .#{$prefix}--data-table tbody th {
-    border-block-end: 1px solid $border-subtle;
+    border-block-end: 1px solid $border-subtle-01;
```

Comment:
Similar to [this PR](https://github.com/carbon-design-system/carbon/pull/13879/files#diff-417d728c7ed2601e1b3cd4667e5c281ea3de98d57769d1919146c42b6ceaf7ac), additional selectors/styles are needed to ensure the token is updated appropriately. `.#{$prefix}--layer-two`, `.#{$prefix}--layer-three`, etc.

Reply:
Without this update the cells will not respond at all to being placed on another layer and will be "locked" to the layer 01 style

Reply:
@tay1orjones I have updated the changes according to your suggestion.